clang/lib/Parse/ParseOpenACC.cpp

```diff
@@ -91,10 +140,19 @@ OpenACCDirectiveKind ParseOpenACCDirectiveKind(Parser &P) {
   P.ConsumeToken();
   std::string FirstTokSpelling = P.getPreprocessor().getSpelling(FirstTok);
 
-  OpenACCDirectiveKind DirKind = getOpenACCDirectiveKind(FirstTokSpelling);
+  OpenACCDirectiveKindEx ExDirKind = getOpenACCDirectiveKind(FirstTokSpelling);
+
+  switch (ExDirKind) {
+  case OpenACCDirectiveKindEx::Invalid:
+    P.Diag(FirstTok, diag::err_acc_invalid_directive) << 0 << FirstTokSpelling;
+    return OpenACCDirectiveKind::Invalid;
+  case OpenACCDirectiveKindEx::Enter:
+  case OpenACCDirectiveKindEx::Exit:
+    return ParseOpenACCEnterExitDataDirective(P, FirstTok, FirstTokSpelling,
+                                              ExDirKind);
+  }
```

Comment:
The switch doesn't actually cover all values, the intent is that `OpenACCDirectiveKindEx` can contain an `OpenACCDirectiveKind`, OR one of the extended values.

So `ExDirKind` can be a bunch of other values, so if the kind is any one of the others (like parallel), it will go through this switch and fall to the below ones.

Reply:
Ah, thats a really good idea!  I was trying to figure out a good way to make it clear what is going on here.